src/cmd2.cc

```diff
@@ -556,58 +545,9 @@ void do_cmd_go_down(void)
 
 		/* Leaving */
 		p_ptr->leaving = TRUE;
-
-		if (!fall_trap)
-		{
```

Comment:
This block shouldn't be removed. Since `fall_trap == FALSE` we have `!fall_trap == TRUE`, so the code should always run.